src/catkin_pkg/topological_order.py

```diff
@@ -85,7 +86,22 @@ def calculate_depends_for_topological_order(self, packages):
                 # avoid function call to improve performance
                 # check within the loop since the set changes every cycle
                 continue
-            packages[name]._add_recursive_run_depends(packages, self.depends_for_topological_order)
+            packages[name]._add_recursive_run_depends_cached(packages, self.depends_for_topological_order)
+
+    def _add_recursive_run_depends_cached(self, packages, depends_for_topological_order):
```

Comment:
Instead of introducing a new method how about updating `_add_recursive_run_depends` instead to do the caching directly?